arraycontext/container/arithmetic.py

```diff
@@ -343,6 +343,17 @@ def {fname}(arg1):
                                 else:
                                     raise ValueError(msg)""")
                     gen(f"return cls({zip_init_args})")
+                if _cls_has_array_context_attr:
+                    gen("if isinstance(arg2,"
+                        "              arg1.array_context.get_array_types()):")
+                    with Indentation(gen):
+                        gen(f"""
+                                return (
+                                    rec_map_array_container(
+                                        lambda a: {op_str.format("a", "arg2")},
+                                        arg1))
```

Comment:
This should use those `_deserialize_init_arrays_code` and `_serialize_init_arrays_code` methods to unwrap the serialization. There should be an example for numpy object arrays lower down (?).

Reply:
I rather just used appended the types in broadcasting logic for numbers.